webapp/packages/plugin-settings-panel/src/SettingsPanel/SettingsList.tsx

```diff
@@ -36,17 +38,17 @@ export const SettingsList = observer<Props>(function SettingsList({ treeData, so
     groups.splice(0, 1, ...treeData.getChildren(groupId));
 
     const group = ROOT_SETTINGS_GROUP.get(groupId)!;
-    const groupSettings = settings.get(group);
 
-    list.push({ group, settings: groupSettings || [] });
+    list.push(group);
```

Comment:
Suggest to move this code to function getGroupList() and do it using for cylcle so there is less chance to get an infinite loop

I'm not a big fan of while cycles